warden/x/warden/keeper/msg_server_new_keychain.go

```diff
@@ -34,3 +39,15 @@ func (k msgServer) NewKeychain(goCtx context.Context, msg *types.MsgNewKeychain)
 		Id: id,
 	}, nil
 }
+
+func ensureKeychainFeesValid(fees *types.KeychainFees) error {
```

Comment:
I think this could become a method of the `*KeychainFees` type, so we can call `fees.Validate()`

(you can add it to the `warden/x/warden/types/v1beta3/keychain.go` file)